mods/gen2/scripts.js

```diff
@@ -117,6 +117,134 @@ exports.BattleScripts = {
+		if (move.ohko) {
+			if (target.speed > pokemon.speed) {
+				this.add('-immune', target, '[ohko]');
+				return false;
+			}
+		}
+
+		// Calculate true accuracy for gen 2, which uses 0-255.
+		if (accuracy !== true) {
+			accuracy = Math.floor(accuracy * 255 / 100);
+			// Check also for accuracy modifiers.
+			if (!move.ignoreAccuracy) {
+				if (pokemon.boosts.accuracy > 0) {
+					accuracy *= boostTable[pokemon.boosts.accuracy];
+				} else {
+					accuracy = Math.floor(accuracy / boostTable[-pokemon.boosts.accuracy]);
+				}
+			}
+			if (!move.ignoreEvasion) {
+				if (target.boosts.evasion > 0 && !move.ignorePositiveEvasion) {
+					accuracy = Math.floor(accuracy / boostTable[target.boosts.evasion]);
+				} else if (target.boosts.evasion < 0) {
+					accuracy *= boostTable[-target.boosts.evasion];
+				}
+			}
+			accuracy = Math.min(accuracy, 255);
+		}
+		accuracy = this.runEvent('Accuracy', target, pokemon, move, accuracy);
+
+		if (accuracy !== true && this.random(255) >= accuracy) {
+			this.attrLastMove('[miss]');
+			this.add('-miss', pokemon);
+			damage = false;
+		}
+
+		// If damage is 0 and not false it means it didn't miss, let's calc.
+		if (damage !== false) {
+			pokemon.lastDamage = 0;
+			if (move.multihit) {
+				let hits = move.multihit;
+				if (hits.length) {
+					// Yes, it's hardcoded... meh
+					if (hits[0] === 2 && hits[1] === 5) {
+						hits = [2, 2, 3, 3, 4, 5][this.random(6)];
+					} else {
+						hits = this.random(hits[0], hits[1] + 1);
+					}
+				}
+				hits = Math.floor(hits);
+				// In gen 1, all the hits have the same damage for multihits move
+				let moveDamage = 0;
+				let firstDamage;
+				let i;
+				for (i = 0; i < hits && target.hp && pokemon.hp; i++) {
+					if (i === 0) {
+						// First hit, we calculate
+						moveDamage = this.moveHit(target, pokemon, move);
+						firstDamage = moveDamage;
+					} else {
+						// We get the previous damage to make it fix damage
+						move.damage = firstDamage;
+						moveDamage = this.moveHit(target, pokemon, move);
+					}
+					if (moveDamage === false) break;
+					damage = (moveDamage || 0);
+					if (target.subFainted) {
+						i++;
+						break;
+					}
+				}
+				move.damage = null;
+				if (i === 0) return true;
+				this.add('-hitcount', target, i);
+			} else {
+				damage = this.moveHit(target, pokemon, move);
+			}
+		}
+
+		if (move.category !== 'Status') {
+			// FIXME: The stored damage should be calculated ignoring Substitute.
+			// https://github.com/Zarel/Pokemon-Showdown/issues/2598
+			target.gotAttacked(move, damage, pokemon);
+		}
+
+		// Checking if substitute fainted
+		if (target.subFainted) doSelfDestruct = false;
+		if (move.selfdestruct && doSelfDestruct) {
+			this.faint(pokemon, pokemon, move);
+		}
+
+		// The move missed.
+		if (!damage && damage !== 0) {
+			// Delete the partial trap lock if necessary.
+			delete pokemon.volatiles['partialtrappinglock'];
```

Comment:
Related to the partial trapping above, it still doesn't work like this is gen 2.